Tests/SentryTests/Integrations/Performance/UIViewController/SentryUIViewControllerSwizzlingTests.swift

```diff
@@ -66,6 +71,50 @@ class SentryUIViewControllerSwizzlingTests: XCTestCase {
         let expectedTransactionName = SentryUIViewControllerSanitizer.sanitizeViewControllerName( controller)
         XCTAssertEqual(expectedTransactionName, transactionName)
     }
+
+    @available(iOS 13.0, *)
+    func testSwizzle_fromScene() {
+        let swizzler = TestSentryUIViewControllerSwizziling(options: fixture.options, dispatchQueue: TestSentryDispatchQueueWrapper())
+        
+        let window = UIWindow()
+        window.rootViewController = TestViewController()
+        let mockWindowScene = ObjectWithWindowsProperty(resultOfWindows: NSSet(array: [window]))
+        
+        let notification = Notification(name: NSNotification.Name(rawValue: "UISceneWillConnectNotification"), object: mockWindowScene)
+        swizzler.swizzleRootViewController(fromSceneDelegateNotification: notification)
+        
+        XCTAssertEqual(swizzler.viewControllers.count, 1)
+        XCTAssertTrue(swizzler.viewControllers[0] is TestViewController)
+    }
+    
+    @available(iOS 13.0, *)
+    func testSwizzle_fromScene_invalideNotification() {
+        let swizzler = TestSentryUIViewControllerSwizziling(options: fixture.options, dispatchQueue: TestSentryDispatchQueueWrapper())
+        
+        let window = UIWindow()
+        window.rootViewController = TestViewController()
+        let mockWindowScene = ObjectWithWindowsProperty(resultOfWindows: window)
+        
+        var notification = Notification(name: NSNotification.Name(rawValue: "UISceneWillConnectNotification"), object: nil)
+        swizzler.swizzleRootViewController(fromSceneDelegateNotification: notification)
+        
+        XCTAssertEqual(swizzler.viewControllers.count, 0)
+        
+        notification = Notification(name: NSNotification.Name(rawValue: "NotUISceneWillConnectNotification"), object: mockWindowScene)
+        swizzler.swizzleRootViewController(fromSceneDelegateNotification: notification)
+
+        XCTAssertEqual(swizzler.viewControllers.count, 0)
+        
+        notification = Notification(name: NSNotification.Name(rawValue: "UISceneWillConnectNotification"), object: "Other type of Object")
+        swizzler.swizzleRootViewController(fromSceneDelegateNotification: notification)
+        
+        XCTAssertEqual(swizzler.viewControllers.count, 0)
+        
+        notification = Notification(name: NSNotification.Name(rawValue: "UISceneWillConnectNotification"), object: ObjectWithWindowsProperty(resultOfWindows: "Windows property of the wrong type"))
+        swizzler.swizzleRootViewController(fromSceneDelegateNotification: notification)
+        
+        XCTAssertEqual(swizzler.viewControllers.count, 0)
```

Comment:
`m`: Nice test, but I think it would make sense to split this up into several test cases with proper namings as this tests many different things.